File: system/src/main/java/com/cys/system/common/service/ArticleService.java
package com.cys.system.common.service;

import com.cys.system.common.common.pojo.Result;
import com.cys.system.common.pojo.Article;
import com.cys.system.common.pojo.Command;
import com.cys.system.common.pojo.TimeTask;
import org.omg.PortableInterceptor.INACTIVE;

import java.io.IOException;
import java.util.List;

public interface ArticleService {
    Result listArticle(Integer page, Integer rows, Article article) throws IOException;

    Result getArticleById(Integer id);

    Result addArticle(Article article);

    Result deleteArticleById(Integer[] ids);

    Result increaseBrowseNum(Integer articleId, Integer userId);

    Result increaseLoveNum(Integer articleId, Integer userId, Integer islove);

    Result commitCommand(Command command) throws Exception;

    Result deleteCommand(Integer commandId, Integer userId);

    void updateStatusById(Integer id);

    List<TimeTask> listTimeTask();

    Result loadArticleList();
}

File: search/src/main/resources/static/search/js/service/articleService.js
app.service("articleService", function ($http) {

    // 根据id查询
    this.findOne = function (id) {
        return $http.get(basePath+"/article/get?id=" + id);
    };
    //文章列表
    this.loadArticleList = function (id) {
        return $http.get(basePath+"/index/loadArticleList");
    };

    // 浏览
    this.browse = function (articleId) {
        return $http.get(basePath+"/article/increaseBrowse?articleId=" + articleId);
    };
    // 喜欢
    this.love = function (articleId, islove) {
        return $http.put(basePath+"/article/increaseLoveNum?articleId=" + articleId + "&islove=" + islove);
    };
    // 添加评论
    this.addCommand = function (command) {
        return $http.post(basePath+"/article/command/commit", command);
    };
    // 删除评论
    this.deleteCommand = function (commandId) {
        return $http.delete(basePath+"/article/command/delete?commandId=" + commandId);
    }
});